tests/test_postal_backend.py

```diff
@@ -0,0 +1,473 @@
+        email.send()
+        data = self.get_api_call_json()
+        self.assertNotIn('plain_body', data)
+        self.assertEqual(data['html_body'], html_content)
+
+    def test_extra_headers(self):
+        self.message.extra_headers = {'X-Custom': 'string', 'X-Num': 123}
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertCountEqual(data['headers'], {
+            'X-Custom': 'string',
+            'X-Num': 123
+        })
+
+    def test_extra_headers_serialization_error(self):
+        self.message.extra_headers = {'X-Custom': Decimal(12.5)}
+        with self.assertRaisesMessage(AnymailSerializationError, "Decimal"):
+            self.message.send()
+
+    @override_settings(ANYMAIL_IGNORE_UNSUPPORTED_FEATURES=True)  # Postal only allows single reply-to
+    def test_reply_to(self):
+        email = mail.EmailMessage('Subject', 'Body goes here', 'from@example.com', ['to1@example.com'],
+                                  reply_to=['reply@example.com', 'Other <reply2@example.com>'])
+        email.send()
+        data = self.get_api_call_json()
+        self.assertEqual(data['reply_to'], 'reply@example.com')  # keeps first email
+
+    def test_multiple_reply_to(self):
+        email = mail.EmailMessage('Subject', 'Body goes here', 'from@example.com', ['to1@example.com'],
+                                  reply_to=['reply@example.com', 'Other <reply2@example.com>'])
+        with self.assertRaises(AnymailUnsupportedFeature):
+            email.send()
+
+    def test_attachments(self):
+        text_content = "* Item one\n* Item two\n* Item three"
+        self.message.attach(filename="test.txt", content=text_content, mimetype="text/plain")
+
+        # Should guess mimetype if not provided...
+        png_content = b"PNG\xb4 pretend this is the contents of a png file"
+        self.message.attach(filename="test.png", content=png_content)
+
+        # Should work with a MIMEBase object (also tests no filename)...
+        pdf_content = b"PDF\xb4 pretend this is valid pdf data"
+        mimeattachment = MIMEBase('application', 'pdf')
+        mimeattachment.set_payload(pdf_content)
+        self.message.attach(mimeattachment)
+
+        self.message.send()
+        data = self.get_api_call_json()
+        attachments = data['attachments']
+        self.assertEqual(len(attachments), 3)
+        self.assertEqual(attachments[0]["name"], "test.txt")
+        self.assertEqual(attachments[0]["content_type"], "text/plain")
+        self.assertEqual(decode_att(attachments[0]["data"]).decode('ascii'), text_content)
+
+        self.assertEqual(attachments[1]["content_type"], "image/png")  # inferred from filename
+        self.assertEqual(attachments[1]["name"], "test.png")
+        self.assertEqual(decode_att(attachments[1]["data"]), png_content)
+
+        self.assertEqual(attachments[2]["content_type"], "application/pdf")
+        self.assertEqual(attachments[2]["name"], "")  # none
```

Comment:
BTW, it's possible that the real Postal API will actually issue an `AttachmentMissingName` error in this case (based on [the docs](https://krystal.github.io/postal-api/controllers/send/message.html#:~:text=AttachmentMissingName)). There are some other ESPs with the same limitation, and Anymail usually just documents that as an ESP-specific quirk.

Reply:
Done!